### PR TITLE
Feat[TE-18438]: Support API application type in Addon creation

### PR DESCRIPTION
**Pull request by SunilGembali**

### CODE REVIEW

File: src/main/java/com/testsigma/sdk/RestAction.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package com.testsigma.sdk;
+
+public abstract class RestAction extends DriverAction{
```
Comment (kasyap-gundu) [inline src/main/java/com/testsigma/sdk/RestAction.java]: @SunilGembali Is this reviewed with Product team.. I feel that We can call this as APIAction. RestAction looks somewhat weird and hard to understand. Please check with them and refactor if needed

File: gradle.properties [hunk 2]
```diff
@@ -3,7 +3,7 @@ RELEASE_SIGNING_ENABLED=true
 
 GROUP=com.testsigma
 POM_ARTIFACT_ID=testsigma-java-sdk
-VERSION_NAME=1.2.11_cloud
+VERSION_NAME=1.2.12_cloud
```
Comment (kasyap-gundu) [inline gradle.properties:6]: @SunilGembali Can we have this as 1.2.12_cloud_beta until qa finishes one round of testing. 
If we find any issues and want to make sdk changes, we need to change the number or version again. It won't be good

**Review by kasyap-gundu — CHANGES_REQUESTED** [review]

File: src/main/java/com/testsigma/sdk/RestAction.java (hunk 1 above)
Comment (kasyap-gundu) [inline src/main/java/com/testsigma/sdk/RestAction.java]: @SunilGembali Should this be extended from DriverAction? RestApi will not have any driver. So please check and change it, we shouldn't be exposing any driver for RestAPI ApplicationType

File: src/main/java/com/testsigma/sdk/ApplicationType.java [hunk 3]
```diff
@@ -1,5 +1,5 @@
 package com.testsigma.sdk;
 
 public enum ApplicationType {
-  WEB,MOBILE_WEB,IOS,ANDROID, WINDOWS,WINDOWS_UFT, MAC
+  WEB,MOBILE_WEB,IOS,ANDROID, WINDOWS,WINDOWS_UFT, MAC, REST
```
Comment (kasyap-gundu) [inline src/main/java/com/testsigma/sdk/ApplicationType.java]: We can refactor REST -> REST_API. 
It'll be understandable for user

**Review by kasyap-gundu — CHANGES_REQUESTED** [review]

File: src/main/java/com/testsigma/sdk/RestAction.java (hunk 1 above)
Comment (SunilGembali) [reply src/main/java/com/testsigma/sdk/RestAction.java]: @rajesh-testsigma  suggested to make it as RestApiAction. WIll change

File: src/main/java/com/testsigma/sdk/ApplicationType.java (hunk 3 above)
Comment (SunilGembali) [reply src/main/java/com/testsigma/sdk/ApplicationType.java]: Changed this one also.

**Review by kasyap-gundu — APPROVED** [review]

**Review by Ganesh-Testsigma — APPROVED** [review]

**Review by kasyap-gundu — APPROVED** [review]

**Review by Ganesh-Testsigma — APPROVED** [review]

**Pull request merged**